app/models/stock.rb

```diff
@@ -1,6 +1,6 @@
 class Stock < ApplicationRecord
-  has_many :portfolio_stocks
-  has_many :orders
+  has_many :portfolio_stocks, dependent: :nullify
+  has_many :orders, dependent: :nullify
```

Comment:
This feels wrong to me because stocks should never be deleted. They should be moved to an `inactive` state instead. Maybe `:restrict_with_error` instead?

Reply:
I like it. 

"should never be deleted" will have a lot more interesting considerations in the future, but this is an easy fix now